### DISCUSSION TITLE
NBXplorer is synchronizing stuck @ 99% although there is enough storage space

### DISCUSSION THREAD

**Discussion by ProTechEx** [btcpayserver/btcpayserver#6407] [Plugins & integrations]
On my My BTCPay Server v2.0.3+558f45a48

The NBXplorer is synchronizing stuck @ 99% although there is enough storage space.

Node purning is set to:
BTCPAYGEN_ADDITIONAL_FRAGMENTS="opt-save-storage-s"

I try to fix it with: export NBXPLORER_BTCRESCAN=1
But still the same...

![image](https://github.com/user-attachments/assets/f5880489-11b6-409a-8b54-a86273ae65df)
![image](https://github.com/user-attachments/assets/e05465c8-e504-43e1-83be-e5d5bfefef51)
![image](https://github.com/user-attachments/assets/dc7ade55-e61d-428b-ac14-8485ed78dabf)

**Comment by mp21-org** [comment]
[Another user has run into the same issue](https://t.me/btcpayserver/81284) since the last btcpayserver/nbxplorer update. The user provided the following logs:

![photo_2024-11-16_15-55-54](https://github.com/user-attachments/assets/fa44a12e-3273-4512-8db8-c9a5d339b389)
![photo_2024-11-16_15-55-50](https://github.com/user-attachments/assets/73b2689b-06ee-4b10-9d56-7b7192c069ef)

**Comment by NicolasDorier** [comment]
@ProTechEx don't use BTCRESCAN, the issue seems to lie down on LTC, not BTC. Will take a look.

**Reply by ProTechEx** [reply]
Okey, how to revert this option back?

**Comment by NicolasDorier** [comment]
It shows that NBX is synching from a state dating from 2 years ago. Did you disabled LTC for 2 years, then turned it on recently?

**Reply by ProTechEx** [reply]
Yes, how to solve that issue?

**Reply by ProTechEx** [reply]
Here NBExplorer Logs:

info: NBXplorer.Indexer.LTC: TCP Connection succeed, handshaking...
info: NBXplorer.Indexer.LTC: Handshaked
info: NBXplorer.Indexer.LTC: Testing RPC connection to http://litecoind:43782/
info: NBXplorer.Indexer.LTC: RPC connection successful
info: NBXplorer.Indexer.LTC: Full node version detected: 210201
info: NBXplorer.Indexer.LTC: NBXplorer is correctly whitelisted by the node
info: NBXplorer.Events: LTC: Node state changed: NotStarted => NBXplorerSynching
info: NBXplorer.Events: BTC: New block 00000000000000000001318a500a3d9c31c4fa50db6f4529692f5636649d66a6 (870818)
info: NBXplorer.Indexer.LTC: Node disconnected (Sync seems stuck after block dc5c39b97cc7677623684b5821700d168c146f6d9cba6350639be6a010e933cb (dc5c                                                                                          39b97cc7677623684b5821700d168c146f6d9cba6350639be6a010e933cb), restarting the connection.)
info: NBXplorer.Events: LTC: Node state changed: NBXplorerSynching => NotStarted
fail: NBXplorer.Indexer.LTC: Unhandled exception in the indexer, retrying in 60 seconds
System.OperationCanceledException: The peer has been disconnected
   at NBXplorer.Backend.Indexer.AskNextHeaders(Node node, CancellationToken token) in /source/NBXplorer/Backend/Indexer.cs:line 382
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /source/NBXplorer/Backend/Indexer.cs:line 215
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /source/NBXplorer/Backend/Indexer.cs:line 220
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /source/NBXplorer/Backend/Indexer.cs:line 148
   at NBXplorer.Backend.Indexer.IndexerLoop() in /source/NBXplorer/Backend/Indexer.cs:line 95
info: NBXplorer.Indexer.LTC: TCP Connection succeed, handshaking...
info: NBXplorer.Indexer.LTC: Handshaked
info: NBXplorer.Indexer.LTC: Testing RPC connection to http://litecoind:43782/
info: NBXplorer.Indexer.LTC: RPC connection successful
info: NBXplorer.Indexer.LTC: Full node version detected: 210201
info: NBXplorer.Indexer.LTC: NBXplorer is correctly whitelisted by the node
info: NBXplorer.Events: LTC: Node state changed: NotStarted => NBXplorerSynching

**Reply by ProTechEx** [reply]
Just fix it with following this instructions:

https://docs.btcpayserver.org/FAQ/General/#stuck-synching-nbxplorer-is-synchronizing-

**Reply by NicolasDorier** [reply]
Just follow the same instruction, but use `LTC` instead of `BTC`.

**Comment by mp21-org** [comment]
My testnode (freshly installed with BTC+LTC+LBTC) synched just fine:
![grafik](https://github.com/user-attachments/assets/a1b0814b-c2ce-4dfe-b9bc-2986fa3a1ae9)

The second user with the issue posted this morning:
> I done it it's working fine on new instance. on the old instance I deleted the nbxexplorer data for ltc from postgresql it did resynch but now I have an issue where I have to rescan the wallet every time there is new transactions

sounds like he messed up his database.
wished they'd post to github directly.

**Reply by NicolasDorier** [reply]
I don't think those two cases are related.
Should open different issue. (not discussion)

**Comment by NicolasDorier** [comment]
Note: I just released a new version of postgres 2.5.14 which increase the command timeout. As such the exception that @petzsch "TimeoutException" shouldn't happen anymore.

I am closing this issue, as there are several issues mixed into one discussion. One issue was just @ProTechEx using `BTC` in the configuration rather than `LTC`. The other issue was related to the timeout  that should be fixed now.
